src/encoded/helpers/upload_credentials.py

```diff
@@ -1,8 +1,88 @@
 import boto3
 import botocore
+import copy
 import json
 
 
+EXTERNAL_BUCKET_STATEMENTS = [
+    {
+        'Action': 's3:GetObject',
+        'Resource': lambda s: 'arn:aws:s3:::%s/*' % s,
+        'Effect': 'Allow',
+    },
+    {
+        'Action': 's3:ListBucket',
+        'Resource': lambda s: 'arn:aws:s3:::%s' % s,
+        'Effect': 'Allow',
+    },
+    {
+        'Action': 's3:GetObjectAcl',
+        'Resource': lambda s: 'arn:aws:s3:::%s/*' % s,
+        'Effect': 'Allow',
+    }
+]
+
+
+def _compile_statements_from_list(buckets_list):
+    statements = []
+    if buckets_list:
+        for ext_policy in EXTERNAL_BUCKET_STATEMENTS:
+            new_policy = copy.copy(ext_policy)
+            new_policy['Resource'] = []
+            for line in buckets_list:
+                line = line.strip()
+                if line:
+                    line = line.strip()
+                    new_policy['Resource'].append(ext_policy['Resource'](line))
+            statements.append(new_policy)
+    return statements
+
+
+def _save_policy_json(policy_json, file_path):
+    with open(file_path + '.json', 'w') as file_handler:
+        json.dump(policy_json, file_handler)
+
+
+def _build_external_bucket_json(file_path):
+    try:
+        with open(file_path) as file_handler:
+            policy_json = {
+                'Version': '2012-10-17',
+                'Statement': [],
+            }
+            buckets_list = [item.strip() for item in file_handler.readlines()]
+            statements = _compile_statements_from_list(buckets_list)
+            if statements:
+                policy_json['Statement'] = statements
+                _save_policy_json(policy_json, file_path)
+    except FileNotFoundError:
```

Comment:
This catches the case where a developer has turned ON external_aws_s3_transfer_allow 
 and set the external_aws_s3_transfer_buckets file, but didn't add a external_aws_s3_transfer_buckets file or add buckets to the external_aws_s3_transfer_buckets.